### PR TITLE
Batch queue + Consolidate responses

### PR DESCRIPTION
**Pull request by Agus**
- [x] Create logic of Order-Batch for multiple wallets
- [x] Create endpoint for getting `batchIds`
- [x] Update endpoint for getting orders with their batch IDs
- [x] Consolidate `buyer app`'s `DataTable` information using wallet grouping

### CODE REVIEW

File: buyer-api/src/services/batchInfo.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+const listBatchIds = async () => listLevelKeys(ordersPerBatch);
+
+/**
+ * @function createBatchId
+ * @returns {Integer} A new Id for the batch of orders
+ */
+// eslint-disable-next-line
+const createBatch = async (payload = []) => {
+  const id = (new Date()).getTime();
```
Comment (manelli) [inline buyer-api/src/services/batchInfo.js]: Use an UUID as the id in order to not generate any confusion (timestamps are very similar)

```js
const uuid = require('uuid/v4');
const id = uuid();
```

Comment (Agus) [reply buyer-api/src/services/batchInfo.js]: I was between uuid/guid and timestamps, with no preferences, I'll change it 👍

File: buyer-api/src/routes/dataOrders/createDataOrder.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import express from 'express';
 import { getOrdersForBuyer } from '../../facades';
 import { asyncError, cache, dataExchange } from '../../utils';
 import signingService from '../../services/signingService';
+import createBatch from '../../services/batchInfo';
```
Comment (nicoayala) [inline buyer-api/src/routes/dataOrders/createDataOrder.js]: Shouldn't this be: `import { createBatch } from '../../services/batchInfo';`? 🤔

File: buyer-api/src/routes/dataOrders/createDataOrder.js [hunk 3]
```diff
@@ -176,9 +176,13 @@ router.post(
     if (errors.length > 0) {
       res.boom.badData('Validation failed', { validation: errors });
     } else {
-      const accounts = await signingService.getAccounts();
-      accounts.map(account => queue.add('createDataOrder', {
+      const { children } = await signingService.getAccounts();
+
+      const batchId = createBatch();
+
+      children.map(account => queue.add('createDataOrder', {
```
[the remark this answers is not in the document]
Comment (Agus) [reply buyer-api/src/routes/dataOrders/createDataOrder.js]: So many mappings, yes, changed 😄

**Comment by Agus** [comment]
Now grouped orders would be under a nice `uuid` 👇 

screen shot 2018-10-09 at 18 13 27

File: buyer-api/src/facades/getBatchesFacade.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ * @param {Object} batchId the data order to store in the cache
+ * @param {Object} batchCache Redis storage used for orders caching
+ * @returns {Promise} Promise which resolves to redis result
+ */
+const addBatchToCache = (batch, batchesCache) =>
+  batchesCache.set(batch.batchId, JSON.stringify(batch), 'EX', batchesTTL);
+
+/**
+ * @async
+ * @function fetchAndCacheDataOrder
```
Comment (nicoayala) [inline buyer-api/src/facades/getBatchesFacade.js]: `fetchAndCacheBatch`

File: buyer-api/src/routes/dataOrders/batches.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ *         description: Leveldb batch Id
+ *         required: true
+ *         type: string
+ *     produces:
+ *       - application/json
+ *     responses:
+ *       200:
+ *         description: When the information could be fetched correctly.
+ *       404:
+ *         description: When the Data Order was not created by the buyer.
```
Comment (nicoayala) [inline buyer-api/src/routes/dataOrders/batches.js]: `When the Batch was not found.`

File: buyer-api/src/services/batchInfo.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ * @function listBatchPairs
+ * @returns {Promise} Promise which resolves to a list of batches.
+ */
+const listBatchPairs = async () => listLevelPairs(ordersPerBatch);
+
+/**
+ * @function createBatchId
+ * @returns {String} A new Id for the batch of orders
+ */
+// eslint-disable-next-line
```
Comment (nicoayala) [inline buyer-api/src/services/batchInfo.js]: What is disabling?

Comment (Agus) [reply buyer-api/src/services/batchInfo.js]: Something old I guess 😅

**Pull request merged**